packages/wrapped-keys-lit-actions/src/lib/common/batchGenerateEncryptedKeys.js

```diff
@@ -0,0 +1,159 @@
+const { encryptPrivateKey } = require('./internal/encryptKey');
+const {
+  generateEthereumPrivateKey,
+} = require('../ethereum/internal/generatePrivateKey');
+const { signMessageEthereumKey } = require('../ethereum/internal/signMessage');
+const {
+  generateSolanaPrivateKey,
+} = require('../solana/internal/generatePrivateKey');
+const { signMessageSolanaKey } = require('../solana/internal/signMessage');
+
+/* global accessControlConditions, actions, Lit*/
+
+async function processEthereumAction(action, ndx) {
+  const { network, generateKeyParams } = action;
+  const messageToSign = action.signMessageParams?.messageToSign;
+
+  const ethereumKey = generateEthereumPrivateKey();
+
+  const [generatedPrivateKey, messageSignature] = await Promise.all([
+    Lit.Actions.runOnce(
+      { waitForResponse: true, name: `encryptEthereumPrivateKey_${ndx}` },
+      async () =>
+        JSON.stringify(
+          await encryptPrivateKey({
+            accessControlConditions,
+            publicKey: ethereumKey.publicKey,
```

Comment:
Why is the pubkey required in the encrypt function since we only encrypt the private key? If we wanna return it in the same object we can just add it separately in the stringify?

Reply:
Because getting the data out of the runOnce requires that we `stringify()` the result (or it comes back as [Object Object]), injecting publicKey back into the result of this function required that I `JSON.parse()` the result, then spread the parsed resulting object back into a new object and inject publicKey into it, only to then JSON.stringify() it back again on the very next line -- very inefficient and tricky to reason about.

This way I get to avoid adding another JSON.parse() to the calling code in the existing network-specific actions :+1: :)

Reply:
No longer valid, we can change this later to move `pubKey` outside the `encryptPrivateKey`